asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -312,30 +336,45 @@ def _dock(
                                 DockingResultCols.POSIT_METHOD.value: oedocking.OEPositMethodGetName(
                                     result.GetPositMethod()
                                 ),
+                                "Pose_ID": i,
                             }
                             posed_ligand.set_SD_data(sd_data)
+                            posed_ligands.append(posed_ligand)
 
                             # Generate info about which target was actually used by multi-reference docking
                             if isinstance(set, DockingInputMultiStructure):
                                 docked_target = set.complexes[result.GetReceptorIndex()]
-                                input_pair = DockingInputPair(
-                                    ligand=set.ligand, complex=docked_target
+                                input_pairs.append(
+                                    DockingInputPair(
+                                        ligand=set.ligand, complex=docked_target
+                                    )
                                 )
                             else:
-                                input_pair = set
-
-                            docking_result = POSITDockingResults(
-                                input_pair=input_pair,
-                                posed_ligand=posed_ligand,
-                                probability=prob,
-                                provenance=self.provenance(),
+                                input_pairs.append(set)
+
+                        # Create Docking Results Objects
+                        docking_results_objects = []
+                        for input_pair, posed_ligand in zip(input_pairs, posed_ligands):
```

Comment:
I think lets just use an explicit branch to seperate out multistructure and single structure logic, even if it is verbose.

No need to unpack and re-pack single pose results, which will slow us down.

Reply:
I am struggling to conceptualize this, so a quick call will help

Reply:
@apayne97 had the right idea and I was confused. No longer relevant.